### PR TITLE
Product-related Bug Fixes, Fine Tuning

### PR DESCRIPTION
**Pull request by sghoweri**
- Fix overall silent JS issue for IE11 in button component
- Temporarily remove band inner `.c-bolt-band` wrapper in Twig template as workaround to band height expansion behavior in IE11
- Adjust video player background layer positioning to address IE11 inconsistencies in sizing of video backgrounds
- Adjust band height animation to use min-height vs height to account for even larger content than the ideal video player size
- Add additional Pattern Lab examples of bands with videos nested inside of bands + icon-only button example
- Rename internal button `style` props key to `color` since `style` is a reserved keyword :)
- Update band animation + workaround to video player event binding to allow for animations to occur independent to if the video player has finished loading or not
- Update button component CSS + internal class added in the component to address small vertical alignment issue of internal button items in IE11
- Adjust max size of iPhone device viewer + remove Macbook box-shadow -- addresses #298 and #296

### CODE REVIEW

File: src/_patterns/04-pages/d8-homepage/band-with-video-test.twig [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    "accountId": "1900410236",
+    "playerId": "r1CAdLzTW"
+  } only %}
+</bolt-band>
+
+  {% embed "@bolt/band.twig" with {
+    size: "small",
+    theme: "xdark",
+    fullBleed: true
+  } only %}
```
Comment (remydenton) [inline src/_patterns/04-pages/d8-homepage/band-with-video-test.twig]: The `only` keyword doesn't work for embeds, right?

File: src/_patterns/02-components/bolt-button/src/button.standalone.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ export class BoltButton extends withComponent(withPreact()) {
   static is = 'bolt-button';
 
   static props = {
-    style: props.string,
+    color: props.string,
```
Comment (remydenton) [inline src/_patterns/02-components/bolt-button/src/button.standalone.js]: What's the rationale for renaming `style` as `color`?  Seems like it creates a little bit of a disconnect between the passed in value (still "style") and the internally used one (now "color").  Not a big deal though.

File: src/_patterns/02-components/brightcove-player/src/brightcove-player.standalone.js [hunk 3]
```diff
@@ -130,10 +130,11 @@ class BrightcoveVideo extends withComponent(withPreact()) {
     // it’s converted to a proper boolean value using JavaScript’s truthiness
     // & falsiness principles.
     // value = Boolean(value);
-    if (value)
+    if (value) {
       this.setAttribute('expandedHeight', value);
-    else
+    } else {
       this.removeAttribute('expandedHeight');
+    }
```
Comment (remydenton) [inline src/_patterns/02-components/brightcove-player/src/brightcove-player.standalone.js]: I like this-- better readability for sure (though maybe you were just fixing linting errors...)

File: src/_patterns/02-components/bolt-button/src/button.standalone.js (hunk 2 above)
Comment (sghoweri) [reply src/_patterns/02-components/bolt-button/src/button.standalone.js]: @remydenton simple: style is a reserved property which confuses older browsers like IE11 when passing in props to the custom element.

I'm still noodling on what we should be naming the 3 layers of theming / coloring / overrides so this was just an internal temporary workaround we get this naming convention more solidified:

## At a high level though:
- Most generic (set at the component level): sets the default component styling (ex. primary vs secondary button styles), inherits from `t-bolt-` theming classes
- More specific: Inherited styling from theme (ex. primary buttons turn indigo on lighter themes, yellow on darker themes)
- Most specific: manually override or force a component to use a particular color theme, ignoring the inherited theming; also includes unique "one-off" theming options that wouldn't come by default (ex. the icon component's "teal" color option)

File: src/_patterns/04-pages/d8-homepage/band-with-video-test.twig (hunk 1 above)
Comment (sghoweri) [reply src/_patterns/04-pages/d8-homepage/band-with-video-test.twig]: *shrug* no? https://twig.symfony.com/doc/2.x/tags/embed.html

I was just copying over a band example from the Product T1 page when pulling this together so I hadn't even given this any thought

File: src/_patterns/02-components/bolt-button/src/button.standalone.js (hunk 2 above)
Comment (remydenton) [reply src/_patterns/02-components/bolt-button/src/button.standalone.js]: Ah, a reserved property!  Makes total sense then.  Thanks for the elaboration on naming conventions-- code reviews are as much for the benefit of the reviewer as the reviewee :)

**Pull request merged**